src/webnn/native/dml/GraphDML.cpp

```diff
@@ -409,30 +365,28 @@ namespace webnn::native ::dml {
             }
         }
         auto TotalTensorSizeInBytes = elementsCount * typeLength;
-        dmlTensorDesc->bufferDesc.DimensionCount = dmlTensorDesc->dimensions.size();
-        dmlTensorDesc->bufferDesc.Sizes = dmlTensorDesc->dimensions.data();
-        dmlTensorDesc->bufferDesc.Strides = dmlTensorDesc->strides.data();
-        dmlTensorDesc->bufferDesc.TotalTensorSizeInBytes = TotalTensorSizeInBytes;
-        dmlTensorDesc->bufferDesc.GuaranteedBaseOffsetAlignment = 0;
-        dmlTensorDesc->bufferDesc.DataType = dataType;
-        dmlTensorDesc->bufferDesc.Flags = tensorFlag;
-
-        dmlTensorsDesc.push_back(dmlTensorDesc);
-        return true;
+        tensorDesc->bufferDesc.DimensionCount = tensorDesc->dimensions.size();
+        tensorDesc->bufferDesc.Sizes = tensorDesc->dimensions.data();
+        tensorDesc->bufferDesc.Strides = tensorDesc->strides.data();
+        tensorDesc->bufferDesc.TotalTensorSizeInBytes = TotalTensorSizeInBytes;
+        tensorDesc->bufferDesc.GuaranteedBaseOffsetAlignment = 0;
+        tensorDesc->bufferDesc.DataType = dataType;
+        tensorDesc->bufferDesc.Flags = tensorFlag;
+
+        mTensorsDesc.push_back(tensorDesc);
+        createdTensorDesc = {DML_TENSOR_TYPE_BUFFER, &tensorDesc->bufferDesc};
+        return {};
     }
 
-    bool CreateDmlTensorDesc(std::vector<std::shared_ptr<DmlTensorDesc>>& dmlTensorsDesc,
-                             const std::shared_ptr<DmlTensorDesc>& dmlTensorDesc,
-                             OperandDescriptor const* desc,
-                             DML_TENSOR_FLAGS tensorFlag = DML_TENSOR_FLAGS::DML_TENSOR_FLAG_NONE) {
+    MaybeError Graph::CreateDmlTensorDesc(DML_TENSOR_DESC& createdTensorDesc,
```

Comment:
Why can't CreateDmlTensorDesc simply return a DML_TENSOR_DESC?

The caller would validate the parameters instead, eliminating the need to to pass a lvalue reference like this.

Reply:
I know what you mean, I intended to return DML_TENSOR_DESC directly. But I find that a wrong DML_TENSOR_DESC will cause caller function crash, such as  CreateOperator(), then gpu will crash, which will cause the CI failure.

Reply:
Could you share the code that caused that crash?

Reply:
It's not easy to share the code now since there are too many changes. For example, if I create a DML_TENSOR_DESC with negative dimension value or some other invalid parameters, it will crash. @bbernhar